state_machine/applications/system-check/tests/imu_test.py

```diff
@@ -0,0 +1,281 @@
+
+    # update result dictionary
+    result_dict["IMU_AccMoving"] = (acc_string, acc_is_moving)
+    return result_dict
+
+
+def rotating_imu_test(result_dict):
+    """
+    Return acceleration, gyro, and magnetic readings from request_imu_data
+    If None, update result dictionary that test was not run
+    Else, check that the norm of the average gyroscope reading is positive
+    and update result dictionary
+    """
+    # record gyro reading
+    prompt = f"""Please rotate the cubesat as best as possible for {wait_time}
+seconds once you start the test.\n"""
+    acc, gyro, mag = request_imu_data(prompt)
+    if gyro is None:
+        # user entered "n", cancel the test
+        result_dict["IMU_GyroRotating"] = (
+            "Rotating test not completed.", None)
+        return result_dict
+
+    # else continue running the test
+    gyro_string = (f"Gyro: {gyro} (deg/s)")
+    gyro_is_rotating = norm(gyro) >= 0.5
+
+    # print result to user
+    if gyro_is_rotating:
+        print("Rotating gyroscope reading is approx. greater than 0 deg/s.")
+    else:
+        print("Rotating gyroscope reading is near 0 deg/s.")
+
+    # update result dictionary
+    result_dict["IMU_GyroRotating"] = (gyro_string, gyro_is_rotating)
+    return result_dict
+
+
+def magnet_imu_test(result_dict):
+    """
+    Return acceleration, gyro, and magnetic readings from request_imu_data
+    If None, update result dictionary that test was not run
+    Else, check that the magnetometer reading increased as the magnet
+    moved closer, and update result dictionary
+    """
+    # record magnetometer reading
+    prompt = f"""Please slowly move the magnet closer to the cubesat for
+{wait_time} seconds once you start the test.\n"""
+    acc, gyro, mag = request_imu_data(prompt)
+    if mag is None:
+        # user entered "n", cancel the test
+        result_dict["IMU_MagMagnet"] = ("Magnet test not completed.", None)
+        return result_dict
+
+    # else continue running the test
+    mag_string = (f"Change in Mag Reading: {mag} (µT)")
+    # for reference, an average fridge magnet has a force of around 1000 µT
+    mag_is_increasing = mag >= 1000
+
+    # print result to user
+    if mag_is_increasing:
+        print("Magnetometer reading is increasing.")
+    else:
+        print("Magnetometer reading is not increasing.")
+
+    # update result dictionary
+    result_dict["IMU_MagMagnet"] = (mag_string, mag_is_increasing)
+    return result_dict
+
+
+def temp_imu_test(result_dict):
+    """
+    Get temperature reading from IMU and ask the user to verify that
+    this reading is correct. Give users a general range for room temperature
+    to help verify, and update result dictionary as per user response
+    """
+    # collect temperature reading, ask user to confirm
+    temp = cubesat.temperature_imu()
+    print(f"IMU Temperature Reading: {temp} degrees Celsius")
+    print(f"Room temperature is generally between {20} and {27} deg Celsius.")
```

Comment:
It's hard to say that the IMU will actually be at room temperature. I suspect that inside the assembled satellite it will be warmer than that since there won't be a ton of air circulation. I would expect it to be somewhere between 20 and 80 C. 

Also, I'd like to cut down the number of user inputs/amount of thinking a person has to do while running this. 

So, let's have this just check that `20 <= temp <= 80` (no user input/confirmation).

Reply:
Sounds good! 

Another note about the temperature sensor -- it consistently returns between 28 and 30 degrees Celsius when I test it out, even though the room is colder than that. I'm not sure if the sensor is incorrect, or if the MCU is generating a little bit extra heat.